applications/asset_tracker/src/main.c

```diff
@@ -425,35 +429,52 @@ static void device_status_send(struct k_work *work)
 		return;
 	}
 
-	if (!atomic_get(&send_data_enable)) {
-		return;
-	}
+	size_t item_cnt = 0;
 
-	ret = modem_info_params_get(&modem_param);
+#ifdef CONFIG_MODEM_INFO
+	int ret = modem_info_params_get(&modem_param);
 
 	if (ret < 0) {
 		printk("Unable to obtain modem parameters: %d\n", ret);
-		return;
+	} else {
+		ret = modem_info_json_object_encode(&modem_param, root_obj);
+		if (ret > 0) {
+			item_cnt = (size_t)ret;
+		}
 	}
+#endif
 
-	len = modem_info_json_object_encode(&modem_param, root_obj);
+	const char *const sensors[] = {
+		CLOUD_CHANNEL_STR_GPS,
+		CLOUD_CHANNEL_STR_FLIP,
+		CLOUD_CHANNEL_STR_TEMP,
+		CLOUD_CHANNEL_STR_HUMID,
+		CLOUD_CHANNEL_STR_AIR_PRESS,
+#if IS_ENABLED(CONFIG_CLOUD_BUTTON)
+		CLOUD_CHANNEL_STR_BUTTON
+#endif
+	};
 
-	if (len < 0) {
-		return;
+	if (service_info_json_object_encode(sensors, ARRAY_SIZE(sensors), NULL,
+					    0, root_obj) == 0) {
+		++item_cnt;
 	}
 
-	device_cloud_data.data.buf = (char *)root_obj;
-	device_cloud_data.data.len = len;
-	device_cloud_data.tag += 1;
+	if (item_cnt > 0) {
```

Comment:
Could inverse the check, do the `cJSON_Delete(root_obj);` and return if true. Then continue with the other stuff with one indent level saved. Not a big deal here with few indents, though.